### ISSUE TITLE
Regression in completions with one possible result

### ISSUE THREAD

**Issue by mqudsi** [fish-shell/fish-shell#7738]
It used to be that completing an argument that evaluates to a single possible result of a directory would first complete `dir` to `dir/` without displaying a list of options on first `<TAB>`/`^I` and then accept that completion and list the contents of the dir as possible completions on second tab. This currently hangs with one completion as the result until the option is accepted, as the completion pager is displayed.

e.g. in the fish source directory:

```fish
~ cat cmake^I
```

No matter how many times tab/^I is pressed, this continues to display a pager with one possible option (`cmake/`) even though the option is accepted (the commandline itself changes from `cmake` to `cmake/`).

I'm unable to bisect it because the behavior in interactive mode (as described) differs from `build/fish -c 'complete -C"nvim cmake"'` which generates multiple entries, but have confirmed it was no present in the last released version of fish.

**Comment by faho** [comment]
It's somewhere around 2b8d2deb0ccc5016a03c3023e0a2709e88c162d1

**Comment by faho** [comment]
Haaaa, it's a smartcase thing:

```fish
complete -c foo -a 'one ONelists' -f
```

reproduces it

**Comment by faho** [comment]
Yeah, let's do the simple fix for now - see that there's only one completion left and bypass the pager.

If we figure out how to display these in the pager later, that can of course go in.

**Comment by krobelus** [comment]
The wrong case in the pager probably needs this fix:

```diff
diff --git a/src/reader.cpp b/src/reader.cpp
index 3a0fd0505..ed768e4a3 100644
--- a/src/reader.cpp
+++ b/src/reader.cpp
@@ -1985,11 +1984,3 @@ bool reader_data_t::handle_completions(const completion_list_t &comp, size_t tok
                 for (idx = 0; idx < max; idx++) {
-                    wchar_t ac = common_prefix.at(idx), bc = el.completion.at(idx);
-                    bool matches = (ac == bc);
-                    // If we are replacing the token, allow case to vary.
-                    if (will_replace_token && !matches) {
-                        // Hackish way to compare two strings in a case insensitive way,
-                        // hopefully better than towlower().
-                        matches = (wcsncasecmp(&ac, &bc, 1) == 0);
-                    }
-                    if (!matches) break;
+                    if (common_prefix.at(idx) != el.completion.at(idx)) break;
                 }
```
On

	$ fish -C "complete -c foo -a 'abc aBd' -f"
	$ foo A^I

before, we would insert the common prefix "ab" - with case taken from the first completion in the list.
The pager uses the prefix, so we get a wrong completion `abd`.
After this change, nothing is inserted - the commandline stays on A after a single Tab and the pager contains the correct completions.

It looks like the incorrect completions in the pager were introduced by 28d67c8f0f4badfea95b0d9f4ed15837fe6e9414.
Before that, whenever there was a common prefix, we would insert it, and not show the pager.
So we'd only show the pager when there was no common prefix to insert.

An alternative fix would be to let each completion keep their prefix - that would be a bit weird though because we'd still want to insert the common one.

**Comment by ridiculousfish** [comment]
Probably this is what my b38a23a46dfac9fdaf453cced61683765790528c was intended to fix, but that got rightly reverted due #7526.

**Comment by krobelus** [comment]
> Probably this is what my b38a23a was intended to fix

Sweet - I've opened a PR to re-roll that commit adding a fix for the spurious escaping.
I haven't tested it very much but it feels reasonable.

(TBH I'm not sure if case-insensitive prefix computation is really that useful - how often do you have multiple valid completion options with matching prefixes but with different case)
